tests/LocalDateTest.php

```diff
@@ -1251,6 +1252,13 @@ public function providerToString() : array
         ];
     }
 
+    public function testCheckWithNullMonthOfYear()
+    {
+        $dayOfMonth = new DayOfMonth();
+
+        $this->assertNull($dayOfMonth->check(31));
```

Comment:
I did not mean just to move the test, but to try and trigger the case when `DayOfMonth::check()` is called with `$monthOfYear === null` when calling `LocalDate::of()`, or `LocalDate::withDay()`.

Then I realized that it's actually never called with a `null` value from anywhere (maybe it should not be nullable then...)

So sorry, the way you did before was OK for now, can you please re-introduce `DayOfMonthTest`, but please move it to a `Field` sub-namespace: `Brick\DateTime\Tests\Field\DayOfMonthTest`.